torch/_higher_order_ops/utils.py

```diff
@@ -873,6 +873,9 @@ def _get_shape_env(
             no_inference_mode_ctx = torch.inference_mode(False)
             ctx_stack.enter_context(new_fake_mode)
             ctx_stack.enter_context(no_inference_mode_ctx)
+            ctx_stack.enter_context(torch._C._DisableFuncTorch())
```

Comment:
Need these to full disable functionalization when runing tests that compose with torch.func.functionalize e.g. test/functorch/test_control_flow.py:test_cond_functionalize

Reply:
What is this function used for? Figuring out the input mutations for gen_schema?

Reply:
> What's wrong with writing a graph pass to run through the subgraphs and follow the alias and the mutation information, without running FakeTensors through it? I remember the previous problem we had was that it was difficult to infer the type of an input, but we can use the type of the inputs we already have.

check_input_alias_and_mutation_return_outputs can be called when the gm is in TorchIR, which doesn't have a schema. So one idea is that: we make_fx the gm to get aten ir first then check the schema to know which operator mutates its input then find all the input storages that gets mutated? Any thoughts?